frontend/src/views/Join.tsx

```diff
@@ -89,6 +88,14 @@ function JoinGamePage() {
       }).catch((err) => reject(err));
   });
 
+  // Get URL query params to determine if the roomId is provided.
+  const urlParams = new URLSearchParams(window.location.search);
+  const roomIdQueryParam = urlParams.get('room');
+  if (!roomId && roomIdQueryParam && isValidRoomId(roomIdQueryParam)) {
```

Comment:
> I think this all looks good. I'll just comment without approving for now since Alan requested changes. Do you think we should always use history rather than Redirect? If so I'll also change the start game redirect I'm working on.

I think the default is to use `history` instead of `Redirect`; however, if you have a good reason for using the latter, that's fine. But yeah, if there isn't any reason, imo `history` is generally better.

Reply:
So this bug was happening because it set the `roomId` to a six-digit number, and then didn't allow you to type anything more because it thought it had already reached max length. I tried moving it into a `useEffect` block with no dependencies (see the image below).

<img width="452" alt="Code into UseEffect Block" src="https://user-images.githubusercontent.com/7987279/95022362-a9610400-062b-11eb-8aec-a2d724cefab7.png">

I fixed the issue by removing the `roomId.length >= 6`, as I don't really think that's necessary IMO and it fixes this problem. If y'all think we should add that back in though, just let me know -- there are other workarounds for this bug I think, like choosing specifically when to set the room ID.

Reply:
I see, that makes sense - if that's the case, another suggestion I have would be to add I think `value={roomId}` (untested) to the input box to make it display the roomId whenever it's set, as right now I think the confusion was that the input box was empty, making you think roomId was unset and you could type in something new. 

^ If that works, I'd still kinda prefer to keep the six-digit maximum haha (mainly cause while testing a random room id it prevents you from overshooting, though tbf idk if that will be relevant to actual end-users though)

Reply:
Yes, based on your below commend I am changing this, which fixes the bug: https://github.com/rocketden/main/pull/50#discussion_r499270955 😄